### PR TITLE
Prepare for release of v0.3.0

### PR DESCRIPTION
**Pull request by nathancoleman**
### Changes proposed in this PR:
Consul API Gateway version being released: `0.3.0`
Now requires:
- consul: `1.12.2`
- consul-k8s: `0.45.0`

### How I've tested this PR:
Compare to #168 

### Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

File: .release/ci.hcl [hunk 1]
```diff
@@ -18,20 +26,22 @@ event "merge" {
 
 event "build" {
   depends = ["merge"]
+
   action "build" {
     organization = "hashicorp"
-    repository = "consul-api-gateway"
-    workflow = "build"
+    repository   = "consul-api-gateway"
+    workflow     = "build"
```
Comment (nathancoleman) [inline .release/ci.hcl:33]: Whitespace is due to initial run of formatter

**Review by mikemorris — APPROVED** [review]
Was https://github.com/hashicorp/consul-api-gateway/pull/186 used for this?

Assuming the `version.go` bump and golden file regeneration won't be needed until _after_ release due to the change in https://github.com/hashicorp/consul-api-gateway/pull/206

**Comment by nathancoleman** [comment]
@mikemorris yes to both 👍

**Review by claire-labry — APPROVED** [review]
Only reviewed the `.release/ci.hcl` file but all makes sense to me! 🚢

**Pull request merged**